### PR TITLE
Ret 2956 Provide something Tribunal asked for - Respond to a notification (ECM) - notify selected parties (email)

### PR DESCRIPTION
**Pull request by paul-peasegood**
Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RET-2956

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/RespondNotificationService.java [hunk 1]
```diff
@@ -32,11 +124,135 @@ public List<DynamicValueType> getSendNotificationSelection(CaseData caseData) {
...
+     * Gets markdown containing the details of the notification responses.
+     * @param sendNotificationType notification
+     * @return markdown of responses
+     */
+    public String getRespondNotificationMarkdown(SendNotificationType sendNotificationType) {
+        var respondNotificationTypeCollection = sendNotificationType.getRespondNotificationTypeCollection();
+        if (respondNotificationTypeCollection == null) {
+            return "";
+        }
+        List<String> resondNotificationMarkdownList =
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/RespondNotificationService.java:156]: respondNotificationMarkdownList (spelling)

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/SendNotificationService.java [hunk 2]
```diff
@@ -127,21 +128,25 @@ public void clearSendNotificationFields(CaseData caseData) {
         caseData.setSendNotificationFullName2(null);
         caseData.setSendNotificationDetails(null);
         caseData.setSendNotificationRequestMadeBy(null);
+        caseData.setSendNotificationEccQuestion(null);
     }
 
+    /**
+     * Gets a list of notifications in the selected format.
+     * @param caseData data to get notifications from
+     * @param format lamda contains details for the formatting
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/SendNotificationService.java:137]: lambda (spelling)

**Review by JackReevies — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/RespondNotificationService.java [hunk 3]
```diff
@@ -2,28 +2,120 @@
...
+    private static final String RESPONSE_DETAILS = "|  | |\r\n"
+        + "| --- | --- |\r\n"
+        + "| RESPONSE %1$S | |\r\n"
+        + "| Response from | %2$s |\r\n"
+        + "| Response date | %3$s |\r\n"
+        + " %4$s\r\n"
+        + "| What's your response to the tribunal? | %5$s\r\n"
+        + "| Do you want to copy correspondence to the other party to satisfy the Rules of Procedure? | %6$s |\r\n";
+
+    private static final String TRIBUNAL = "TRIBUNAL";
```
[the remark this answers is not in the document]
Comment (paul-peasegood) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/RespondNotificationService.java:64]: added to et-common

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/SendNotificationService.java [hunk 4]
```diff
@@ -96,6 +96,8 @@ public void createSendNotification(CaseData caseData) {
         sendNotificationType.setSendNotificationFullName2(caseData.getSendNotificationFullName2());
         sendNotificationType.setSendNotificationDetails(caseData.getSendNotificationDetails());
         sendNotificationType.setSendNotificationRequestMadeBy(caseData.getSendNotificationRequestMadeBy());
+        sendNotificationType.setSendNotificationEccQuestion(caseData.getSendNotificationEccQuestion());
+        sendNotificationType.setSendNotificationWhoCaseOrder(caseData.getSendNotificationWhoCaseOrder());
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/SendNotificationService.java:100]: Was this meant to be judgment?

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/RespondNotificationServiceTest.java [hunk 5]
```diff
@@ -140,22 +158,90 @@ void testGetRespondNotificationMarkdownAllFields() {
...
-                + "| Additional Information | info |\r\n"
-                + "| Description | title |\r\n"
-                + " | Document | <a href=\"/documents/6fbf9470-f735-484a-9790-5b246b646fe2/binary\""
-                + " target=\"_blank\">TEST.PDF</a>\r\n"
-                + "| Case management order made by | Judge |\r\n"
-                + "| Name | John Doe |\r\n"
-                + "| Sent to | Both parties |\r\n";
+            + "| --- | --- |\r\n"
+            + "| Subject | title |\r\n"
+            + "| Notification | [Hearing, Case management orders / requests, Judgment] |\r\n"
```
[the remark this answers is not in the document]
Comment (paul-peasegood) [reply src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/RespondNotificationServiceTest.java:163]: This is a list, so it does appar like this on the page

[the remark this answers is not in the document]
Comment (JackReevies) [reply src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/RespondNotificationServiceTest.java:281]: According to the figma design, its outside brackets. Just `String.join` the array here to remove that

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/RespondNotificationServiceTest.java [hunk 6]
```diff
@@ -140,22 +276,90 @@ void testGetRespondNotificationMarkdownAllFields() {
...
+            + "| Hearing | 1: Hearing - Barnstaple - 16 May 2022 01:00 |\r\n"
+            + "| Date Sent | 01-JAN-1970 |\r\n"
+            + "| Sent By | TRIBUNAL  |\r\n"
```
Comment (JackReevies) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/RespondNotificationServiceTest.java:284]: The figma shows this as "Tribunal" not "TRIBUNAL". QA will deffo raise this and it does look a little bit....aggressive? on the actual page

**Review by JackReevies — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/RespondNotificationService.java [hunk 7]
```diff
@@ -32,11 +128,145 @@ public List<DynamicValueType> getSendNotificationSelection(CaseData caseData) {
...
+     */
+    public String getRespondNotificationMarkdown(SendNotificationType sendNotificationType) {
+        var respondNotificationTypeCollection = sendNotificationType.getRespondNotificationTypeCollection();
+        if (respondNotificationTypeCollection == null) {
+            return "";
+        }
+        List<String> respondNotificationMarkdownList =
+            respondNotificationTypeCollection.stream().map(respondNotificationType -> {
+                Integer index =
+                    respondNotificationTypeCollection.indexOf(respondNotificationType) + 1;
```
Comment (Itay-Kirsh) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/RespondNotificationService.java:163]: fori loop might be nicer

**Review by Itay-Kirsh — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/RespondNotificationService.java (hunk 7 above)
Comment (paul-peasegood) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/RespondNotificationService.java:165]: In this case, I think the stream is fine

**Review by Itay-Kirsh — APPROVED** [review]

**Review by JackReevies — APPROVED** [review]

**Pull request merged**